dia-batching-server/src/api/binance.rs

```diff
@@ -0,0 +1,115 @@
+use crate::api::error::BinanceError;
+use crate::types::{AssetSpecifier, Quotation};
+use chrono::Utc;
+use rust_decimal::prelude::Zero;
+use rust_decimal::Decimal;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+pub struct BinancePriceApi {
+	client: BinanceClient,
+}
+
+impl BinancePriceApi {
+	pub fn new() -> Self {
+		let client: BinanceClient = BinanceClient::default();
+
+		Self { client }
+	}
+
+	pub async fn get_price(&self, asset: &AssetSpecifier) -> Result<Quotation, BinanceError> {
+		let binance_asset_identifier = Self::convert_to_binance_id(&asset);
+
+		match self.client.price(binance_asset_identifier).await {
+			Ok(price) => Ok(Quotation {
+				symbol: asset.symbol.clone(),
+				name: asset.symbol.clone(),
+				blockchain: Some(asset.blockchain.clone().into()),
+				price: price.price,
+				supply: Decimal::zero(),
+				time: Utc::now().timestamp().unsigned_abs(),
+			}),
+			Err(error) => {
+				log::warn!("Error getting price for {:?} from Binance: {:?}", asset, error);
+				Err(BinanceError(format!("Could not get price for {:?}", asset)))
```

Comment:
Why to create a new `BinanceError` here? We may loose granularity on the error returned by `price`. Not strong opinion though.

Reply:
True, changed it